feat(treesitter): add 'injection.self' and 'injection.parent' (#24831)
neovim/neovim: runtime/doc/news.txt

  • `@injection.language` now has smarter resolution and will now fallback to language aliases and/or attempt lower case variants of the text.
    language via aliases (e.g., filetype) registered via
    `vim.treesitter.language.register`.

>   • The `injection.self` and `injection.parent` keys can be set with the `#set!` directive to apply either the current node's language,
>     or the parent LanguageTree's language, respectively.

Reviewer: ```suggestion
  • The `#set!` directive now supports `injection.self` and `injection.parent` for injecting either the current node's language
    or the parent LanguageTree's language, respectively. 
```
